main.py

```diff
@@ -41,9 +41,10 @@ def email_cloud_function(event, context):
 def process_reminder(event, context):
     if 'cron_schedule' in event:
         cron_schedule = event['cron_schedule']
+        timezone = event.get('timezone')
         event_timestamp = parser.parse(context.timestamp)
         normalized_timestamp = event_timestamp.replace(second=0, microsecond=0)
```

Comment:
I think there's going to be a similar bug for the check_day_of_week() and check_ndays_schedule() code. Let's fix it by moving the timezone conversion into this function, so that normalized_timestamp is in the right timezone. Then let's pass that into all the callee functions, instead of reprocessing context.timestamp for each